src/support/small_set.h

```diff
@@ -34,10 +34,78 @@
 
 namespace wasm {
 
-template<typename T, size_t N, typename FlexibleSet> class SmallSetBase {
+template<typename T, size_t N> struct FixedStorageBase {
+  size_t used = 0;
+  std::array<T, N> storage;
+};
+
+template<typename T, size_t N>
+struct UnorderedFixedStorage : public FixedStorageBase<T, N> {
+  void insert(const T& x) {
+    assert(this->used < N);
+    this->storage[this->used++] = x;
+  }
+
+  void erase(const T& x) {
+    for (size_t i = 0; i < this->used; i++) {
+      if (this->storage[i] == x) {
+        // We found the item; erase it by moving the final item to replace it
+        // and truncating the size.
+        this->used--;
+        this->storage[i] = this->storage[this->used];
+        return;
+      }
+    }
+  }
+};
+
+template<typename T, size_t N>
+struct OrderedFixedStorage : public FixedStorageBase<T, N> {
+  void insert(const T& x) {
+    assert(this->used < N);
+
+    if (this->used == 0) {
+      this->storage[0] = x;
+      this->used = 1;
+      return;
+    }
```

Comment:
No need for this special case, I think.

Reply:
Good catch... not sure why I thought that was needed. Done.